src/lib/linked-list.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const Node = require('./node');
+
+module.exports = class LinkedList {
+    constructor() {
+        this.length = 0;
+        this.head = null;
+        this.tail = null;
+    }
+
+    push(value) { //Big O(n)
+        const node = new Node(value);
+        console.log(node);
+
+        if(!this.head) {
+            this.head = node;
+            this.tail = node;
+            this.length++;
+            return this;
+        }
+
+        let currentNode = this.head;
+        while(currentNode.next) {
+            currentNode = currentNode.next;
+        }
+        currentNode.next = node;
+        this.tail = currentNode.next;
+        this.length++;
+
+        return this;
+    }
+
+    pop() { //Big O(n)
+        if(this.length === 0) {
+            return undefined;
+        }
+
+        let currentNode = this.head;
+        if (!currentNode.next) {
+            this.head = null;
+            this.tail = null;
+            this.length--;
+            return currentNode;
+        }
+        while(currentNode.next) {
+            if (currentNode.next === this.tail) {
+                this.tail = currentNode;
+                currentNode.next = null;
+                this.length--;
+                return this;
+            }
+            currentNode = currentNode.next;
+        }
+
+        return this;
+    }
+
+    remove(value) { //Big O(n)
+        if (!this.head) {
+            return undefined;
+        }
+        let currentNode = this.head;
+        while(currentNode.next) {
+            if (currentNode.value === value) {
+                this.head = this.head.next;
+                this.length--;
+                return this;
+            }
+            if (currentNode.next.value === value) {
+                if (!currentNode.next.next) {
+                    this.tail = currentNode;
+                }
+                currentNode.next = currentNode.next.next;
+                this.length--;
+                return this;
+            }
+            currentNode = currentNode.next;
+        }
+
+        return this;
+    }
+
+};
+
+    Object.prototype.map = function(callback) { //Big O(n)
+    if (this === null) { //Linked List cannot be null
+        return undefined;
+    }
+    if (typeof callback !== 'function') { //the callback parameter has to be a function
+        return undefined;
+    }
+    if (arguments.length > 1) { //identifies additional parameters if they exist
+        var arg = arguments[1];
+    }
+
+    //Converts the Linked List values into an array
+    let currentNode = this.head;
+    let tempArray = [this.head.value];
+    for (let i = 1; currentNode.next; i++) {
+        tempArray[i] = currentNode.next.value;
+        currentNode = currentNode.next;
+    }
+
+    Object.defineProperty(tempArray, counter, { //allows for values to be overwritten
```

Comment:
Huge kudos for doing something pretty complex with `Object.defineProperty` for this problem, but I don't think it was hugely necessary for its scope. Nevertheless, still a great job configuring this.